packages/teams-js/src/public/media.ts

```diff
@@ -350,12 +367,137 @@ export namespace media {
    */
   export interface AudioProps {
     /**
-     * Optional; the maximum duration in minutes after which the recording should terminate automatically.
+     * Optional; the maximum duration in minutes after which the recording should terminate automatically
      * Default value is defined by the platform serving the API.
      */
     maxDuration?: number;
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Base class which holds the callback and notifies events to the host client
+   */
+  abstract class MediaController<T> {
+    protected controllerCallback: T;
+
+    public constructor(controllerCallback?: T) {
+      this.controllerCallback = controllerCallback;
+    }
+
+    protected abstract getMediaType(): MediaType;
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * This function will be implemented by the respective media class which holds the logic
+     * of specific events that needs to be notified to the app.
+     * @param mediaEvent indicates the event signed by the host client to the app
+     */
+    protected abstract notifyEventToApp(mediaEvent: MediaControllerEvent): void;
+
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent): Promise<void>;
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * @deprecated
+     * As of 2.0.0-beta.2, please use {@link media.notifyEventToHost media.notifyEventToHost(mediaEvent: MediaCallerEvent): Promise\<void\>} instead.
+     *
+     * Function to notify the host client to programatically control the experience
+     * @param mediaEvent indicates what the event that needs to be signaled to the host client
+     */
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void);
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
+      ensureInitialized(FrameContexts.content, FrameContexts.task);
+
+      const wrappedFunction: InputFunction<void> = () =>
+        new Promise<void>(resolve => {
+          const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
+          if (err) {
+            throw err;
+          }
+
+          const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
+          resolve(sendAndHandleSdkError('media.controller', [params]));
+        });
+
+      return callCallbackWithErrorOrResultFromPromiseAndReturnPromise<void>(wrappedFunction, callback);
+    }
+
+    public stop(): Promise<void>;
+    /**
+     * Function to programatically stop the ongoing media event
+     * Optional; @param callback is used to send app if host client has successfully stopped the event or not
+     */
+    public stop(callback?: (err?: SdkError) => void): void;
```

Comment:
Does this method definition compile??
The moment you pass the callback to the function notifyEventToHost, it will resolve to the return type void instead of Promise<void>
```
public stop(): Promise<void>;
    public stop(callback: (err?: SdkError) => void): void;
    public stop(callback?: (err?: SdkError) => void): Promise<void> {
      return Promise.resolve(this.notifyEventToHost(MediaControllerEvent.StopRecording, callback));
    }
```
  }

Reply:
I'm still not sure what you mean by this. I thought regardless of the callback, it should call the notifyEventToHost() which returns a promise.

Reply:
@ndangudubiyyam If you haven't already, can you please follow up with @aeunms as soon as possible so we can figure out if we need to open a bug and/or fix this before merging.

Reply:
I thought we are reverting the promise code to v1 version